### DISCUSSION TITLE
Best way to load rtx in non-interactive shells?

### DISCUSSION THREAD

**Discussion by AlecRust** [jdx/mise#734] [General]
There seems to be two documented ways of loading rtx in environments that don't load `~/.zshrc` file:

Either add the shims to PATH:

```sh
export PATH="$HOME/.local/share/rtx/shims:$PATH"
```

Or load rtx then run `hook-env` to update PATH (with non-shim binaries I guess):

```sh
eval "$(rtx activate bash)"
eval "$(rtx hook-env)"
```

Out of curiosity which is the recommended approach? Wasn't totally clear to me from the docs.

It seems to me if the second approach achieves the same end result just without shims, it's probably better?

**Edit:** See: https://github.com/jdx/mise-docs/pull/14

**Comment by Pilgrim1379** [comment]
I don't know about "recommended approach" but I use the second approach and it's worked fine for me so far.

**Reply by AlecRust** [reply]
Yep both work. I've [switched](https://github.com/AlecRust/dotfiles/commit/51dce678ae3c6d0b0226a0e25487fb1b29658b28) to shims as that is the recommended approach in this non-interactive case, see https://github.com/jdx/mise-docs/pull/14.

**Reply by jdx** [reply]
it's a soft recommendation though. The main thing is if you change directory inside of a script it won't work unless you call `mise hook-env` again but with shims that won't be necessary. If you don't care about that there isn't a reason to switch.

The only downside (I can think of) of shims in scripts is something that is probably pretty unlikely. If you wanted to detect if a tool is missing, you could do that with `hook-env` because it wouldn't point to anything whereas you could have a shim that points to an inactive version so `which node` and `type node` would work but actually calling it would fail.
